pkg/apis/apps/validation/validation_test.go

```diff
@@ -42,6 +42,96 @@ func intStrAddr(intOrStr intstr.IntOrString) *intstr.IntOrString {
 	return &intOrStr
 }
 
+type setTweak func(s apps.StatefulSet)
+
+func makeSet(selectorLabels *map[string]string, template *api.PodTemplate, tweaks ...setTweak) apps.StatefulSet {
+	s := apps.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{Name: "abc", Namespace: metav1.NamespaceDefault},
+		Spec: apps.StatefulSetSpec{
+			PodManagementPolicy: apps.OrderedReadyPodManagement,
+			Selector:            &metav1.LabelSelector{MatchLabels: *selectorLabels},
```

Comment:
This should produce a fully validate-safe object - I feel like you're maybe missing fields?  Isn't replicas required?  I see it missing in some test cases, so maybe not.

Reply:
Replicas occur in 17/74 stateful set test cases, so I think maybe it is better to create a tweak func for it other than set a default value.